packages/assets-controllers/src/NftController.ts

```diff
@@ -1402,65 +1412,79 @@ export class NftController extends BaseControllerV1<NftConfig, NftState> {
     userAddress?: string;
     networkClientId?: NetworkClientId;
   }) {
-    const chainId = this.getCorrectChainId({ networkClientId });
+    const releaseLock = await this.mutex.acquire();
 
-    const nftsWithChecksumAdr = nfts.map((nft) => {
-      return {
-        ...nft,
-        address: toChecksumHexAddress(nft.address),
-      };
-    });
-    const nftMetadataResults = await Promise.allSettled(
-      nftsWithChecksumAdr.map(async (nft) => {
-        const resMetadata = await this.getNftInformation(
-          nft.address,
-          nft.tokenId,
-          networkClientId,
-        );
+    try {
+      const time = Date.now();
+      const timeSinceLastNftUpdateRequest = time - this.lastNftRefreshTime;
+
+      if (timeSinceLastNftUpdateRequest < RATE_LIMIT_NFT_INTERVAL) {
+        return;
+      }
+
+      const chainId = this.getCorrectChainId({ networkClientId });
+
+      const nftsWithChecksumAdr = nfts.map((nft) => {
         return {
-          nft,
-          newMetadata: resMetadata,
+          ...nft,
+          address: toChecksumHexAddress(nft.address),
         };
-      }),
-    );
-    const successfulNewFetchedNfts = nftMetadataResults.filter(
-      (result): result is PromiseFulfilledResult<NftUpdate> =>
-        result.status === 'fulfilled',
-    );
-    // We want to avoid updating the state if the state and fetched nft info are the same
-    const nftsWithDifferentMetadata: PromiseFulfilledResult<NftUpdate>[] = [];
-    const { allNfts } = this.state;
-    const stateNfts = allNfts[userAddress]?.[chainId] || [];
-
-    successfulNewFetchedNfts.forEach((singleNft) => {
-      const existingEntry: Nft | undefined = stateNfts.find(
-        (nft) =>
-          nft.address.toLowerCase() ===
-            singleNft.value.nft.address.toLowerCase() &&
-          nft.tokenId === singleNft.value.nft.tokenId,
+      });
+      const nftMetadataResults = await Promise.allSettled(
```

Comment:
no need for allSettled here`getNftInformation` is already handling async errors by using `safelyExecute`